R/utils.R

```diff
@@ -1004,3 +1004,36 @@ clean_hugo_cache = function() {
 yes_no = function(question) {
   interactive() && tolower(substr(readline(paste(question, '(y/n) ')), 1, 1)) == 'y'
 }
+
+# Will output like this
+# If choices is named, the name is used as suffix as in example below
+# Title:
+# ---------
+# 1: value (name)
+# ---------
+# footnote
+# ask
+select_choice <- function(choices, title, footnote, ask, width = getOption("width")) {
+  if (!missing(title)) {
+    cat(format(title, width = width), sep = "\n")
+    cat(hrule(width = width), sep = "\n")
+  }
+  index = seq_along(choices)
+  if (!is.null(nm <- names(choices))) {
+    for (j in nm) {
+      if (!is.na(j) && nzchar(j))
+        choices[j] <- sprintf("%s (%s)", choices[j], j)
+    }
+  }
+  cat(format(paste(index, choices, sep = ": "), width = width), sep = "\n")
+  cat(hrule(width = width), sep = "\n")
+  if (!missing(footnote)) cat(format(footnote, width = width), sep = "\n")
+  if (missing(ask)) {
+    ask = "Use a number above to select (type ESC twice to cancel): "
```

Comment:
I think `Esc` only works in RStudio. For other applications, you probably need to press `Ctrl + C`.